Add doc on new field `request.kubernetes_resources` (#48480)
gravitational/teleport: docs/pages/includes/role-spec.mdx

      # resources accessible by the listed roles (enterprise-only)
      search_as_roles: ['access']

      # 'kubernetes_resources' restricts what kinds of Kubernetes resources

>       # a user can access request to. The example defined below, enforces users to 

Reviewer: ```suggestion
      # a user can request access to. In the below example, users can
```

>       # a user can access request to. The example defined below, enforces users to 

Reviewer 2: ```suggestion
      # a user can access request to. The example defined below enforces users to 
```

      # request only Kubernetes namespaces. Default (when nothing is defined) allows

>       # any Kubernetes resource request.

Reviewer 3: ```suggestion
      # access requests to any Kubernetes resource or the entire cluster.
```
